packages/backend/src/invoice/service.test.ts

```diff
@@ -0,0 +1,220 @@
+        }
+      }
+    )
+
+    afterEach(
+      async (): Promise<void> => {
+        await truncateTable(knex, 'invoices')
+      }
+    )
+
+    test('Defaults to fetching first 20 items', async (): Promise<void> => {
+      const invoices = await invoiceService.getUserInvoicesPage(user.id)
+      expect(invoices).toHaveLength(20)
+      expect(invoices[0].id).toEqual(invoicesCreated[0].id)
+      expect(invoices[19].id).toEqual(invoicesCreated[19].id)
+      expect(invoices[20]).toBeUndefined()
+    })
+
+    test('Can change forward pagination limit', async (): Promise<void> => {
+      const pagination = {
+        first: 10
+      }
+      const invoices = await invoiceService.getUserInvoicesPage(
+        user.id,
+        pagination
+      )
+      expect(invoices).toHaveLength(10)
+      expect(invoices[0].id).toEqual(invoicesCreated[0].id)
+      expect(invoices[9].id).toEqual(invoicesCreated[9].id)
+      expect(invoices[10]).toBeUndefined()
+    })
+
+    test('Can paginate forwards from a cursor', async (): Promise<void> => {
+      const pagination = {
+        after: invoicesCreated[19].id
+      }
+      const invoices = await invoiceService.getUserInvoicesPage(
+        user.id,
+        pagination
+      )
+      expect(invoices).toHaveLength(20)
+      expect(invoices[0].id).toEqual(invoicesCreated[20].id)
+      expect(invoices[19].id).toEqual(invoicesCreated[39].id)
+      expect(invoices[20]).toBeUndefined()
+    })
+
+    test('Can paginate forwards from a cursor with a limit', async (): Promise<void> => {
+      const pagination = {
+        first: 10,
+        after: invoicesCreated[9].id
+      }
+      const invoices = await invoiceService.getUserInvoicesPage(
+        user.id,
+        pagination
+      )
+      expect(invoices).toHaveLength(10)
+      expect(invoices[0].id).toEqual(invoicesCreated[10].id)
+      expect(invoices[9].id).toEqual(invoicesCreated[19].id)
+      expect(invoices[10]).toBeUndefined()
+    })
+
+    test("Can't change backward pagination limit on it's own.", async (): Promise<void> => {
+      const pagination = {
+        last: 10
+      }
+      const invoices = invoiceService.getUserInvoicesPage(user.id, pagination)
+      await expect(invoices).rejects.toThrow(
+        "Can't paginate backwards from the start."
+      )
+    })
+
+    test('Can paginate backwards from a cursor', async (): Promise<void> => {
+      const pagination = {
+        before: invoicesCreated[20].id
+      }
+      const invoices = await invoiceService.getUserInvoicesPage(
+        user.id,
+        pagination
+      )
+      expect(invoices).toHaveLength(20)
+      expect(invoices[0].id).toEqual(invoicesCreated[0].id)
+      expect(invoices[19].id).toEqual(invoicesCreated[19].id)
+      expect(invoices[20]).toBeUndefined()
+    })
+
+    test('Can paginate backwards from a cursor with a limit', async (): Promise<void> => {
+      const pagination = {
+        last: 10,
+        before: invoicesCreated[10].id
```

Comment:
Using a later cursor or a smaller limit would better test the limit.
As is, you'd get the same result without `last`.

Reply:
This test was being overloaded to check that a backwards paginated output results in the same as a forwards pagination. So I split that out into another test too.